OPT_mission.Tanoa/sectorcontrol/functions/fnc_initplayereh.sqf

```diff
@@ -76,7 +82,7 @@ if (GVAR(flagFreeMineZoneOn)) then {
                 if ((typeOf _explosive) find "SatchelCharge" != -1 or (typeOf _explosive) find "DemoCharge" != -1) exitWith {};    
 
                 // only if near flag
-                if ({_explosive distance _x <= GVAR(flagFreeMineZoneRadius)} count (GVARMAIN(nato_flags) + GVARMAIN(csat_flags)) == 0) exitWith {};
+                if ({_explosive distance (_x select 0) <= GVAR(flagFreeMineZoneRadius)} count (GVARMAIN(nato_flags) + GVARMAIN(csat_flags)) == 0) exitWith {};
```

Comment:
Hier könnte man sich das `_x select 0` auch sparen, wenn man die Flaggen nicht jeweils nochmal in ein extra Array verpackt :)

Reply:
Schön wie detailliert Du das alles durchgehst und Dir das alles auffällt. Gute Arbeit :)
Ich bin wegen dieser Array Sache offen für Verbesserungsvorschläge. Habe es leider nicht besser hinbekommen, da ich ArmA in der Hinsicht nicht verstanden habe, da es anscheinend unmöglich ist ein Array mit nur einem einzigen Element zu füllen. (Für den Fall das es nur einen Fahne gibt)

Reply:
Y pushBack [z] für ein Element  z=[Y] .  Block in Block  [Y] append [Z] ist dann z =[[y]]

Reply:
Wie Lord schon geschrieben hat kannst du dir für Array-Manipulationen die Befehle `append`und `pushBack` anschauen. Mit denen kann man da eig alles machen :)

https://community.bistudio.com/wiki/append
https://community.bistudio.com/wiki/pushBack

> a es anscheinend unmöglich ist ein Array mit nur einem einzigen Element zu füllen. (Für den Fall das es nur einen Fahne gibt)

Das ist ohne Probleme möglich. Was hast du denn versucht, das nicht geklappt hat? :)

Reply:
Eines der größten Probleme war, das das an vielen Stellen vorkommende Array-zusammensetzende `GVARMAIN(nato_flags) + GVARMAIN(csat_flags)` (mit + oder append) nichts mehr zurückgibt, wenn eins der beiden Arrays komplett leer ist.

Einen minimalistischen Test-Codeschnipsel zum Rumspielen kann ich heute Abend mal raussuchen.

Reply:
Also `append` gibt nie etwas zurück. Das erstellt nämlich kein neues Array, sondern modifiziert das bestehende.

`+` hingegen sollte eigentlich den gewünschten Effekt haben. Das Problem, das du da beschreibst hab ich so noch nie gehört. Auf der anderen Seite ist Arma alles zuzutrauen xD

Reply:
Anbei ein einfaches Beispiel. Gegeben sei:
`private _vieleObjekte = nearestObjects [position player, [], 100]; 
private _einObjekt = _vieleObjekte select 0; 
private _keinObjekt = [];`

Versucht man nun
`private _alleObjekte = _keinObjekt + _einObjekt + _vieleObjekte;`
oder
`_keinObjekt append _einObjekt;`,
wird ein "Generic error in expression" geworfen. Als Fehlerstelle wird das erste "+" oder das "append" gezeigt.

Reply:
_einObjekt ist in dem Fall auch kein array sonder nur ein element. du zieht ja auch den Array _vieleObjekte mit _vieleObjekte select 0 das erste element heraus. Und somit geht das dann nicht. Wenn das element wieder als einzelen Array hinein stellen willst , dann muss da ne Klammer herum. _keinObjekt append [_einObjekt];

Reply:
Bzw für den Fall noch besser: `_keinObjekt pushBack _einObjekt`

Wenn du allerdings den `+` operator verwenden willst, dann musst du (wie Lord schon richtig gesagt hat) sowas hier machen:

```
private _alleObjekte = _keinObjekt + [_einObjekt] + _vieleObjekte;
```
Die Erklärung warum, hat Lord ja schon gegeben :point_up:

Reply:
mit dem Trick ,kann man auch eine Einzeles Element als Array in eine anderen Array packen 
`_Gebäude append [_Haus]; _Gebäude append []; _Gebäude append _Gebäudearten;`   Sieht das so aus. _Gebäude => [[_Haus],[],[Villa,Hochaus,Hotel]];

Reply:
Okay, aber das ist doch im Prinzip genau das, was ich hier im PR an vielen Stellen schon angewendet habe und der Auslöser für diese Diskussion war - oder verstehe ich das gerade falsch?

Reply:
Habe ich nicht wo gelesen, das du Probleme hast einen leeren Array zu übertragen?

Reply:
Bisher wurden diese Variablen ja nur für eine einzige Flagge verwendet. Durch die Erweiterung hier sind nun aber auch Keine oder >1 Flaggen möglich, deshalb muss es meiner Meinung nach ein Array sein.

Reply:
> oder verstehe ich das gerade falsch?

Ich glaube schon. Dein Problem bestand darin, dass du einzelne Objekte als Arrays verwenden wolltest, obwohl sie keine Arrays sind. In den Fällen, in denen du das brauchst (und ich sehe noch nicht ganz an welcher Stelle genau das benötigt wird), kannst du den Trick verwenden, dass du das Ganze in ein Array verpackst, indem du `[]` verwendest.
Allerdings solltest du die Flaggen nicht als Arrays in einem Array speichern, da es sich bei den Flaggen nunmal nicht um Arrays handelt. Daher ergibt es auch nicht viel Sinn sie als Array zu speichern.

tl;dr: Benutze `pushBack` um die Flaggen als `[Flagge1, Flagge2, ...]` zu speichern und an den Stellen, an denen du tatsächlich ein Array aus einer einzelnen Flaggen machen musst, benutze `[]` lokal an dieser Stelle.

> Bisher wurden diese Variablen ja nur für eine einzige Flagge verwendet. Durch die Erweiterung hier sind nun aber auch Keine oder >1 Flaggen möglich, deshalb muss es meiner Meinung nach ein Array sein.

Du hast ja auch ein Array (die `csat_flags` oder wie die Variable heißt). Nur in diesem Array solltest ud die Flaggen nicht nochmal als Array verpacken. Das ist meines aktuellen Verständnisses der Sachlage nicht nötig :point_up: